src/Features/LanguageServer/Protocol/CustomProtocol/FindUsagesLSPContext.cs

```diff
@@ -0,0 +1,291 @@
+            _metadataAsSourceFileService = metadataAsSourceFileService;
+
+            CancellationToken = cancellationToken;
+        }
+
+        public async override Task OnCompletedAsync()
+        {
+            var referencesToReport = ArrayBuilder<VSReferenceItem>.GetInstance();
+            using (await _semaphore.DisposableWaitAsync(CancellationToken).ConfigureAwait(false))
+            {
+                if (_resultsChunk.IsEmpty())
+                {
+                    referencesToReport.Free();
+                    return;
+                }
+
+                referencesToReport.AddRange(_resultsChunk.ToArray());
+                _resultsChunk.Clear();
+            }
+
+            // We can report outside of the lock here since _progress is thread-safe.
+            _progress.Report(referencesToReport.ToArrayAndFree());
+        }
+
+        public async override Task OnDefinitionFoundAsync(DefinitionItem definition)
+        {
+            var referencesToReport = ArrayBuilder<VSReferenceItem>.GetInstance();
+
+            using (await _semaphore.DisposableWaitAsync(CancellationToken).ConfigureAwait(false))
+            {
+                if (_definitionToId.ContainsKey(definition))
+                {
+                    return;
+                }
+
+                // Assigning a new id to the definition
+                _id++;
+                _definitionToId.Add(definition, _id);
+
+                // VSReferenceItem currently doesn't support the ClassifiedTextElement type for DefinitionText,
+                // so for now we just pass in a string.
+                // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/918138
+                var classifiedText = definition.GetClassifiedText();
+
+                var pooledBuilder = PooledStringBuilder.GetInstance();
+                foreach (var text in classifiedText.Runs)
+                {
+                    pooledBuilder.Builder.Append(text.Text);
+                }
+
+                var definitionText = pooledBuilder.ToStringAndFree();
+
+                // Creating a new VSReferenceItem for the definition
+                var definitionItem = await GenerateVSReferenceItemAsync(
+                    _id, definitionId: _id, _document, _position, definition.SourceSpans.FirstOrDefault(),
+                    definition.DisplayableProperties, _metadataAsSourceFileService, definitionText,
+                    symbolUsageInfo: null, CancellationToken).ConfigureAwait(false);
+
+                if (definitionItem != null)
+                {
+                    AddToReferencesToReport(referencesToReport, definitionItem);
+                }
+            }
+
+            if (referencesToReport.IsEmpty())
+            {
+                referencesToReport.Free();
+                return;
+            }
+
+            // We can report outside of the lock here since _progress is thread-safe.
+            _progress.Report(referencesToReport.ToArrayAndFree());
```

Comment:
```suggestion
            _progress.Report(referencesToReport.ToArray());
```

Or remove .ToArray if .Report takes an IEnumerable.